arch/arm/src/s32k3xx/hardware/s32k3xx_emios.h

```diff
@@ -0,0 +1,320 @@
+#define EMIOS_C_DMA                  (1 << 24) /* Bit 24: Direct Memory Access (DMA) */
+#define EMIOS_C_UCPREN               (1 << 25) /* Bit 25: Prescaler Enable (UCPREN) */
+#define EMIOS_C_UCPRE_SHIFT          (26)      /* Bits 26-27: Prescaler (UCPRE) */
+#define EMIOS_C_UCPRE_MASK           (0x03 << EMIOS_C_UCPRE_SHIFT)
+#  define EMIOS_C_UCPRE_DIV1         (0x00 << EMIOS_C_UCPRE_SHIFT) /* Divide by 1 */
+#  define EMIOS_C_UCPRE_DIV2         (0x01 << EMIOS_C_UCPRE_SHIFT) /* Divide by 2 */
+#  define EMIOS_C_UCPRE_DIV3         (0x02 << EMIOS_C_UCPRE_SHIFT) /* Divide by 3 */
+#  define EMIOS_C_UCPRE_DIV4         (0x03 << EMIOS_C_UCPRE_SHIFT) /* Divide by 4 */
+
+#define EMIOS_C_ODISSL_SHIFT         (28)      /* Bits 28-29: Output Disable Select (ODISSL) */
+#define EMIOS_C_ODISSL_MASK          (0x03 << EMIOS_C_ODISSL_SHIFT)
+#  define EMIOS_C_ODISSL_IN0         (0x00 << EMIOS_C_ODISSL_SHIFT) /* Output Disable Input 0 */
+#  define EMIOS_C_ODISSL_IN1         (0x01 << EMIOS_C_ODISSL_SHIFT) /* Output Disable Input 1 */
+#  define EMIOS_C_ODISSL_IN2         (0x02 << EMIOS_C_ODISSL_SHIFT) /* Output Disable Input 2 */
+#  define EMIOS_C_ODISSL_IN3         (0x03 << EMIOS_C_ODISSL_SHIFT) /* Output Disable Input 3 */
+
+#define EMIOS_C_ODIS                 (1 << 30) /* Bit 30: Output Disable (ODIS) */
+#define EMIOS_C_FREN                 (1 << 31) /* Bit 31: Freeze Enable (FREN) */
+
+/* UC Status n (Sn) */
+
+#define EMIOS_S_FLAG                 (1 << 0)  /* Bit 0: Flag indicating input capture or match event (FLAG) */
+#define EMIOS_S_UCOUT                (1 << 1)  /* Bit 1: UC Output Pin state (UCOUT) */
+#define EMIOS_S_UCIN                 (1 << 2)  /* Bit 2: UC Input Pin state (UCIN) */
+                                               /* Bits 3-14: Reserved */
+#define EMIOS_S_OVFL                 (1 << 15) /* Bit 15: Overflow (OVFL) */
+                                               /* Bits 16-30: Reserved */
+#define EMIOS_S_OVR                  (1 << 31) /* Bit 31: Overrun (OVR) */
+
+/* Alternate Address n (ALTAn) */
+
+#define EMIOS_ALTA_SHIFT             (0)       /* Bits 0-15: Alternate Address */
+#define EMIOS_ALTA_MASK              (0xffff << EMIOS_ALTA_SHIFT)
+                                               /* Bits 16-31: Reserved */
+
+/* UC Control 2 n (C2_n) */
+
+#define EMIOS_C2_UCRELDEL_INT_SHIFT  (0)       /* Bits 0-4: Reload Signal Output Delay Interval */
+#define EMIOS_C2_UCRELDEL_INT_MASK   (0x1f << EMIOS_C2_UCRELDEL_INT_SHIFT)
+                                               /* Bits 5-13: Reserved */
+#define EMIOS_C2_UCPRECLK            (1 << 14) /* Bit 14: Prescaler Clock Source (UCPRECLK) */
+                                               /* Bit 15: Reserved */
+#define EMIOS_C2_UCEXTPRE_SHIFT      (16)      /* Bits 16-19: Extended Prescaler (UCEXTPRE) */ 
+#define EMIOS_C2_UCEXTPRE_MASK       (0x0f << EMIOS_C2_UCEXTPRE_SHIFT)
+#define EMIOS_C2_UCEXTPRE(n)         ((n << EMIOS_C2_UCEXTPRE_SHIFT) & EMIOS_C2_UCEXTPRE_MASK)
```

Comment:
```suggestion
#define EMIOS_C2_UCEXTPRE(n)         (((n) << EMIOS_C2_UCEXTPRE_SHIFT) & EMIOS_C2_UCEXTPRE_MASK)
```
here and other places in this file